pkg/cmd/ssh/ssh_patch_options.go

```diff
@@ -0,0 +1,343 @@
+
+	if clientConfig, err := manager.ClientConfig(ctx, gardenTarget); err == nil {
+		if rawConfig, err := clientConfig.RawConfig(); err == nil {
+			context, ok := rawConfig.Contexts[rawConfig.CurrentContext]
+			if !ok {
+				return nil, fmt.Errorf("Could not determine current authInfo. Failed to read current context from kubeconfig")
+			}
+
+			authInfo, ok = rawConfig.AuthInfos[context.AuthInfo]
+			if !ok {
+				return nil, fmt.Errorf("Could not determine current authInfo. Failed to read auth info/user from kubeconfig")
+			}
+		} else {
+			return nil, fmt.Errorf("Could not retrieve raw config: %v", err)
+		}
+	} else {
+		return nil, fmt.Errorf("Could not retrieve client config for target garden %s: %v", sshTarget.GardenName(), err)
+	}
+
+	return authInfo, nil
+}
+
+func getBastionsOfUser(ctx context.Context, gardenClient gardenClient.Client, authInfo *api.AuthInfo) ([]*operationsv1alpha1.Bastion, error) {
+	bastionOfUser := []*operationsv1alpha1.Bastion{}
+
+	currentUser, err := getCurrentUser(ctx, gardenClient, authInfo)
+	if err != nil {
+		return nil, err
+	} else if currentUser == "" {
+		return bastionOfUser, nil
+	}
+
+	list, err := gardenClient.ListBastions(ctx)
+	if err != nil || len(list.Items) == 0 {
+		return nil, err
+	}
+
+	for i := range list.Items {
+		bastion := list.Items[i]
+		if createdBy, ok := bastion.Annotations["gardener.cloud/created-by"]; ok {
+			if createdBy == currentUser {
+				bastionOfUser = append(bastionOfUser, &bastion)
+			}
+		}
+	}
+
+	return bastionOfUser, nil
+}
+
+func getBastionNamesOfUser(ctx context.Context, gardenClient gardenClient.Client, authInfo *api.AuthInfo) ([]string, error) {
+	bastions, err := getBastionsOfUser(ctx, gardenClient, authInfo)
+	if err != nil {
+		return []string{}, err
+	}
+
+	names := make([]string, len(bastions))
+	for i, bastion := range bastions {
+		names[i] = bastion.Name
+	}
+
+	return names, nil
+}
+
+func patchBastion(ctx context.Context, o *SSHPatchOptions, gardenClient gardenClient.Client, sshTarget target.Target) error {
+	bastion, err := getBastion(ctx, o, gardenClient, sshTarget)
+	if err != nil {
+		return err
+	}
+
+	oldBastion := bastion.DeepCopy()
+
+	hasChanges, _, err := patchBastionResource(o, bastion)
+
+	if err != nil {
+		return err
+	} else if !hasChanges {
+		return nil
+	}
+
+	return gardenClient.PatchBastion(ctx, bastion, oldBastion)
+}
+
+func getGardenClient(f util.Factory) (target.Manager, target.Target, gardenClient.Client, error) {
+	manager, err := f.Manager()
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	// sshTarget is the target used for the run method
+	sshTarget, err := manager.CurrentTarget()
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	gardenClient, err := manager.GardenClient(sshTarget.GardenName())
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	return manager, sshTarget, gardenClient, nil
+}
+
+func (o *SSHPatchOptions) Run(f util.Factory) error {
+	_, sshTarget, gardenClient, err := getGardenClient(f)
```

Comment:
In the long run do not want to pass the factory anymore to the `Run` function, hence for new commands we shouldn't use it anymore.
Instead, put everything you need for the `Run` function on the options struct.